docs/guide/tensorboard.rst

```diff
@@ -225,3 +225,38 @@ Here is an example of how to render an episode and log the resulting video to Te
     model = A2C("MlpPolicy", "CartPole-v1", tensorboard_log="runs/", verbose=1)
     video_recorder = VideoRecorderCallback(gym.make("CartPole-v1"), render_freq=5000)
     model.learn(total_timesteps=int(5e4), callback=video_recorder)
+
+
+Directly Accessing The Summary Writer
+-------------------------------------
+
+If you would like to log arbitrary data (in one of the formats supported by torch.utils.tensorboard), you 
+can get direct access to the underlying SummaryWriter in a callback:
+
+.. warning::
+    This is method is not recommended and should only be used by advanced users.
+
+.. code-block:: python
+
+    from stable_baselines3 import SAC
+    from stable_baselines3.common.callbacks import BaseCallback
+
+
+    model = SAC("MlpPolicy", "Pendulum-v0", tensorboard_log="/tmp/sac/", verbose=1)
+
+
+    class SummaryWriterCallback(BaseCallback):
+        def _on_training_start(self):
+            output_formats = self.logger.Logger.CURRENT.output_formats
+            # Save reference to tensorboard formatter object
+            # note: the failure case (not formatter found) is not handled here, should be done with try/except.
+            self.tb_formatter = next(formatter for formatter in output_formats if isinstance(formatter, TensorBoardOutputFormat))
+
+        def _on_step(self) -> bool:
+            if self.n_calls % self._render_freq == 0:
+                self.tb_formatter.writer.add_text("direct_access", "this is a value", self.n_calls)
```

Comment:
`self.n_calls` or `self.n_timesteps`?

btw, did you test that code?

Reply:
I usually use n_calls and it is what is used in [other parts of the docs](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html?highlight=callback#using-callback-monitoring-training) ? I tested it but oviously but simplified it a bit too much before copying it to the docs. I'll test the exact code Ill push in the next commit.

Reply:
Updated and fully tested the example in ffacc5f8af4c77cfea7b1b9c24ecbd7bb3bbbc4d .

Reply:
so yes, n calls is used for checkpointing, and n timesteps is usually used for logging (to tensorboard notably)